homeassistant/components/forecast_solar/coordinator.py

```diff
@@ -0,0 +1,163 @@
+"""The Forecast.Solar integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+import random
+from datetime import datetime, timedelta
+
+import async_timeout
+from aiohttp import ClientResponseError
+from forecast_solar import (
+    Estimate,
+    ForecastSolar,
+    ForecastSolarConnectionError,
+    ForecastSolarError,
+    ForecastSolarRatelimit,
+)
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt
+
+from .const import (
+    LOGGER,
+    DOMAIN
+)
+
+
+class ForecastSolarUpdateCoordinator(DataUpdateCoordinator[Estimate]):
+    """Update coordinator for Forecast.Solar."""
+
+    def __init__(
+            self,
+            hass: HomeAssistant,
+            logger: logging.Logger,
+            api: ForecastSolar,
+            api_interval: timedelta,
+    ) -> None:
+        """Initialize."""
+        super().__init__(
+            hass,
+            logger,
+            name=DOMAIN,
+            update_interval=timedelta(minutes=1),
+            update_method=self.update,
+        )
+        self.api = api
+        # TODO In case there are multiple instances, they should share the api_interval
```

Comment:
Why is there a TODO here?

Reply:
Ah, should have clarified this! Because this is a limitation of the forecast solar library. The homeassistant interface to it doesn't / shouldn't know the details of the API limits. This issue was present before already, I just wanted to make it explicit.

The issue is the following: If you create two forecast solar instances (e.g. for a west and east side of a roof) both of them will query once per hour. I am assuming that forecast solar is rate-limiting based on IP, so one IP would effectively query every thirty minutes, exceeding the rate limit.

Reply:
I reformulated it